elegant/segment_images.py

```diff
@@ -140,6 +147,47 @@ def annotate_poses_from_masks(positions, mask_root, annotations, overwrite_exist
                 current_annotation[annotation] = pose
             current_annotation[original_annotation] = pose
 
+def gmm_lawn_maker(image, optocoupler, return_model=False):
+    '''Find a lawn in an image use Gaussian mixture modeling (GMM)
+
+    This lawn maker models an image (i.e. its pixel intensities) as as mixture
+        of two Gaussian densities. Each corresponds to either the background & lawn.
+
+    Parameters:
+        image - numpy ndarray of the image to find the lawn from
+        optocoupler - optocoupler magnification (as a float) used for the specified image
+
+    Returns:
+        lawn mask as a bool ndarray
+        fitted GMM model
+    '''
+    filtered_image = filters.median_filter(image, size=(3,3), mode='constant')
+    vignette_mask = process_images.vignette_mask(optocoupler, image.shape)
+
+    img_data = filtered_image[vignette_mask]
+    img_hist = numpy.bincount(img_data.flatten())
+    img_hist = img_hist/img_hist.sum()
+
+    gmm = mixture.GaussianMixture(n_components=2)
+    gmm.fit(numpy.expand_dims(img_data,1))
+
+    # Calculate boundary point for label classification as intensity threshold
+    gmm_support = numpy.linspace(0,2**16-1,2**16)
+
+    labels = gmm.predict(numpy.reshape(gmm_support, (-1,1)))
+    thr = numpy.argmax(numpy.abs(numpy.diff(labels)))
+    lawn_mask = (filtered_image < thr) & vignette_mask
+
+    lawn_mask = morphology.binary_erosion(lawn_mask, iterations=10)
+    lawn_mask = zpl_mask.get_largest_object(lawn_mask)
```

Comment:
should this maybe be the last function run? The dilation could join two objects, which I imagine you might want joined, and then you grab the largest contiguous object as a last step? Or is it actually better to grab a single object at the beginning?

Actually, I'm a little confused at what's going on. Usually you erode and then dilate to get rid of small objects, but you're already grabbing just the largest object. Or is the erosion and dilation to smooth the final object?